src/pybind/mgr/dashboard/frontend/src/app/shared/models/osd-deployment-options.ts

```diff
@@ -0,0 +1,24 @@
+export enum OsdDeploymentOptions {
+  COST_CAPACITY = 'cost_capacity',
+  THROUGHPUT = 'throughput_optimized',
+  IOPS = 'iops_optimized'
+}
+
+export interface DeploymentOption {
+  name: OsdDeploymentOptions;
+  capacity: number;
+  available: boolean;
+  hdd_used: number;
+  used: number;
+  nvme_used: number;
+  ssd_used: number;
+  option_name: string;
+  msg: string;
+}
```

Comment:
I'd love to see something like that. There are tools like [this](https://github.com/phillipdupuis/pydantic-to-typescript), which simply convert pydantic models to Typescript interfaces. Also [pydantic supports built-in JSONSchema generation](https://pydantic-docs.helpmanual.io/usage/schema/), which [can later be converted to Typescript classes](https://www.npmjs.com/package/json-schema-to-typescript).

But, yeah, pydantic is not yet supported by CentOS 8, right? Well, we don't need "runtime" support, since this conversion would only happen at "build time" or not even so (a developer manually invokes the scripts to convert from Py to TS).